src/rebar3_codecov_prv.erl

```diff
@@ -53,20 +59,44 @@ format_error(Reason) ->
 analyze(Files) ->
     try
         cover:start(),
-        lists:map(fun(F) ->  cover:import(F),
-                             rebar_api:info("importing ~s~n", [F]) end, Files),
+        lists:map(fun(F) ->
+                      cover:import(F),
+                      rebar_api:info("importing ~s~n", [F])
+                  end,
+                  Files),
         Modules = cover:imported_modules(),
         {result, Result, _} = cover:analyse(Modules, calls, line),
         Result
-    catch Error:Reason ->
-              rebar_api:abort("~p~n~p~n~p~n",[Error, Reason, erlang:get_stacktrace()])
+    catch
+        Error:Reason ->
+            rebar_api:abort("~p~n~p~n~p~n",[Error, Reason, erlang:get_stacktrace()])
     end.
 
-to_json(Data) ->
+export(Data, State) ->
     Mod2Data = lists:foldl(fun add_cover_line_into_array/2, #{}, Data),
+    Formats = export_formats(State),
+    to_json(Mod2Data, Formats),
+    to_lcov(Mod2Data, Formats).
+
+export_formats(State) ->
+    {Args, _} = rebar_state:command_parsed_args(State),
+    LCov = proplists:get_value(lcov, Args, false),
+    JSON = proplists:get_value(json, Args, true),
+    #{lcov => LCov, json => JSON}.
+
+to_json(Mod2Data, #{json := true}) ->
+    rebar_api:info("exporting ~s~n", [?JSON_OUT_FILE]),
     JSON = maps:fold(fun format_array_to_list/3, [], Mod2Data),
     Binary = jiffy:encode(#{<<"coverage">> => {JSON}}),
-    file:write_file(?OUT_FILE, Binary).
+    file:write_file(?JSON_OUT_FILE, Binary);
+to_json(_, _) -> ok.
+
+to_lcov(Mod2Data, #{lcov := true}) ->
+    rebar_api:info("exporting ~s~n", [?LCOV_OUT_FILE]),
+    {ok, LCovFile} = file:open(?LCOV_OUT_FILE, [write]),
+    maps:fold(fun module_to_lcov/3, LCovFile, Mod2Data),
+    file:close(LCovFile);
+to_lcov(_, _) -> ok.
```

Comment:
There is a some duplication here, the map seems a bit unnecessary and maybe just returning a list like `[lcov]` or `[lcov, json]` from `export_formats/1` could be more intuitive. We could then just iterate over the list with a function that has one clause for `lcov` and one for `json`. It's just an observation, no mandatory changes here.

Reply:
See no real difference. In my opinion map is a bit simpler to use when we have some default value for the settings which can be overwritten.

Reply:
No functional difference indeed, only a few lines shorter and without a short-lived map. Anyway, the current version is OK. We might change it if we ever get more formats.